src/modules/auth/token.service.ts

```diff
@@ -95,22 +99,33 @@ export class TokenService {
       token = req.cookies?.token;
     }
 
-    if (token) {
-      const decodedToken = jwt.decode(token) as TokenPayload;
-      const isBrowserRequestFromAuthenticatedOrigin =
-        decodedToken?.origin === req.headers['origin'];
-      const isServerRequestOrGETFromSameDomain =
-        req.headers['origin'] === undefined;
-      if (
-        isBrowserRequestFromAuthenticatedOrigin ||
-        isServerRequestOrGETFromSameDomain
-      ) {
-        next();
-      } else {
-        throw new UnauthorizedException('Origins not matched');
-      }
-    } else {
-      throw new UnauthorizedException('Unauthorized');
+    if (!token) {
+      throw new UnauthorizedException('Access token was not set');
+    }
+
+    const decodedToken = jwt.decode(token) as TokenPayload;
+    this.matchOriginAgainstRequest(decodedToken?.origin, req);
+
+    next();
+  }
+
+  /**
+   * Checks that `origin` of token corresponds to origin of request
+   * @param origin Origin specified for token
+   * @param req Http request object being authenticated with token
+   */
+  matchOriginAgainstRequest(origin: string, req: Request) {
+    const isBrowserRequestFromAuthenticatedOrigin =
+      origin === req.headers['origin'];
+    const isServerRequestOrGETFromSameDomain =
+      req.headers['origin'] === undefined;
+    if (
+      !isBrowserRequestFromAuthenticatedOrigin &&
+      !isServerRequestOrGETFromSameDomain
+    ) {
+      throw new UnauthorizedException(
+        `Token origin ${origin} does not matches request origin ${req.headers['origin']}`
+      );
```

Comment:
As I think this is a bit separate from the direct CORS configuration and is a refactoring rather than new functionality, I think it would be better to have this in a separate PR. We can leave it here, but just a point to keep in mind for future.

Reply:
Yes, it is not directly relates to CORS. I separated matching origin to token in order not to repeat it separately for access and refresh token